File: pertemuan10/Prak/PRAK10_2473021_PY/C_NilaiPredikat.py
"""
Pertemuan 10
[PRAKTIKUM]  program yang meminta input N data untuk Nama dan IPK mahasiswa untuk disimpan ke dalam suatu list.

@author 2473021-Febrianus Leona Putra
@version 26 November 2024
"""
n = int(input("N: "))

arr1 = []  
arr2 = []  
for i in range(n):  
    nama = (input("Nama : "))  
    ipk = float(input("IPK : "))  
    arr1.append(nama)
    arr2.append(ipk)

print("-------------------------------")

ipkTertinggi = max(arr2)
indeksTertinggi = arr2.index(ipkTertinggi)
namaTertinggi = arr1[indeksTertinggi]

dengan_pujian = []
sangat_memuaskan = []
memuaskan = []

for i in range(n):
    if arr2[i] > 3.5:
        dengan_pujian.append(arr1[i])
    elif arr2[i] > 3.0:
        sangat_memuaskan.append(arr1[i])
    elif arr2[i] > 2.75:
        memuaskan.append(arr1[i])

print(f"IPK tertinggi: {namaTertinggi}: {ipkTertinggi}")
print("Predikat dengan Pujian: ", end="")
for i, nama in enumerate(dengan_pujian):
    if i > 0:
        print(", ", end="")
    print(nama, end="")
print()  
print("Predikat Sangat Memuaskan: ", end="")
for i, nama in enumerate(sangat_memuaskan):
    if i > 0:
        print(", ", end="")
    print(nama, end="")
print()  
print("Predikat Memuaskan: ", end="")
for i, nama in enumerate(memuaskan):
    if i > 0:
        print(", ", end="")
    print(nama, end="")
print()

File: pertemuan10/Prak/PRAK10_2473021_JAVA/C_NilaiPredikat.java
/**
 * Pertemuan 10
 * [PRAKTIKUM]  program yang meminta sc N data untuk Nama dan IPK mahasiswa untuk disimpan ke dalam suatu list.
 *
 * @author 2473021-Febrianus Leona Putra
 * @version 26 November 2024
 */
package pertemuan10.Prak.PRAK10_2473021_JAVA;

import java.util.ArrayList;
import java.util.Scanner;

public class C_NilaiPredikat {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("N: ");
        int n = sc.nextInt();
        sc.nextLine();

        ArrayList<String> arr1 = new ArrayList<>();
        ArrayList<Double> arr2 = new ArrayList<>();

        for (int i = 0; i < n; i++) {
            System.out.print("Nama : ");
            String nama = sc.nextLine();

            System.out.print("IPK : ");
            double ipk = sc.nextDouble();
            sc.nextLine();

            arr1.add(nama);
            arr2.add(ipk);
        }

        System.out.println("-------------------------------");

        double ipkTertinggi = arr2.get(0);
        int indeksTertinggi = 0;
        for (int i = 1; i < arr2.size(); i++) {
            if (arr2.get(i) > ipkTertinggi) {
                ipkTertinggi = arr2.get(i);
                indeksTertinggi = i;
            }
        }
        String namaTertinggi = arr1.get(indeksTertinggi);

        ArrayList<String> denganPujian = new ArrayList<>();
        ArrayList<String> sangatMemuaskan = new ArrayList<>();
        ArrayList<String> memuaskan = new ArrayList<>();

        for (int i = 0; i < n; i++) {
            double ipk = arr2.get(i);
            String nama = arr1.get(i);
            if (ipk > 3.5) {
                denganPujian.add(nama);
            } else if (ipk > 3.0) {
                sangatMemuaskan.add(nama);
            } else if (ipk > 2.75) {
                memuaskan.add(nama);
            }
        }

        System.out.println("IPK tertinggi: " + namaTertinggi + ": " + ipkTertinggi);
        System.out.print("Predikat dengan Pujian: ");

        for (int i = 0; i < denganPujian.size(); i++) {
            if (i > 0) {
                System.out.print(", ");
            }
            System.out.print(denganPujian.get(i));
        }

        System.out.println();

        System.out.print("Predikat Sangat Memuaskan: ");

        for (int i = 0; i < sangatMemuaskan.size(); i++) {
            if (i > 0) {
                System.out.print(", ");
            }
            System.out.print(sangatMemuaskan.get(i));
        }

        System.out.println();

        System.out.print("Predikat Memuaskan: ");

        for (int i = 0; i < memuaskan.size(); i++) {
            if (i > 0) {
                System.out.print(", ");
            }
            System.out.print(memuaskan.get(i));
        }

        System.out.println();

        sc.close();
    }
}
